shell.nix

```diff
@@ -0,0 +1,10 @@
+(import
```

Comment:
Not sure if it’s possible but is there any way that `shell.nix` and `default.nix` can instead just be included within `flake.nix`? Or do they really need to be three separate files?

Reply:
Unfortunately, the community is pretty divided on flakes, and there's not much integration between the people using them and the people not using them. The flake is the modern way to do things, and makes things significantly easier, while the `default` is the old way of doing things, using the flake as a source of truth.

However, I *can* move everything (or at the very least `default`, haven't tested past that before) to a `nix` directory, losing some ergonomics, as well as put the logic for the shell in the `default.nix`. Not sure why I had made the shell instead of doing that, I think I was confused because I'm always confused while using Nix haha.

TL;DR, I'll move the shell into the `default.nix` at least, do you want me to try moving everything in a `nix` directory? I'm not sure the `flake.nix` *can*, but I can try, I know the `default.nix` can.

Reply:
Oh I see, no worries then! Let's just keep them as .nix files in the root directory.

Reply:
Update: `flake.nix` can, but it makes using it as a Nix flake a bit more convoluted / less expected (e.g., instead of `nix run "github:MilesCranmer/rip2"` it'll be `nix run "github:MilesCranmer/rip2?dir=nix"`). It's a usability concern at most though, as far as I can tell. Up to y'all, what do you think?

Reply:
I think I would just go with what most people do. I'm assuming people will just see there is a .nix file here and then run the command they always run on the github URL.